Unable to vsphere.newClient

from Gitter

@sdorsett juin 26 16:57
the error didn't make it into the above post:
Jun 26 09:52:32 cds-engine cds-engine[5777]: 2020-06-26 09:52:32 [ERROR] hatchery:vsphere> Serve: Create> Init error: Unable to vsphere.newClient: Post "https:///sdk": http: no Host in request URL

@sdorsett juin 27 06:01
as a test I updated /opt/cds/conf.toml to have an empty api.vsphere.endpoint and restarted the hatchery. I got the expected error:
Jun 26 22:58:35 cds-engine cds-engine[6388]: Unable to init service hatchery:vsphere: vsphere-endpoint is mandatory
'hatchery.vsphere.endpoint' that was.

@sdorsett juin 27 06:19
so it looks like the /opt/cds/conf.toml hatchery.vsphere values are not being properly passed to vsphere.newClient

--- a/engine/hatchery/vsphere/init.go
+++ b/engine/hatchery/vsphere/init.go
@@ -13,30 +13,27 @@ import (
 
 // InitHatchery create new client for vsphere
 func (h *HatcheryVSphere) InitHatchery(ctx context.Context) error {
-
 	h.user = h.Config.VSphereUser
 	h.password = h.Config.VSpherePassword
 	h.endpoint = h.Config.VSphereEndpoint
 
 	// Connect and login to ESX or vCenter
-	c, errNc := h.newClient(ctx)
-	if errNc != nil {
-		return fmt.Errorf("Unable to vsphere.newClient: %s", errNc)
+	c, err := h.newClient(ctx)
+	if err != nil {
+		return fmt.Errorf("Unable to vsphere.newClient: %v", err)
 	}
 	h.vclient = c
 
 	finder := find.NewFinder(h.vclient.Client, false)
 	h.finder = finder
 
-	var errDc error
-	if h.datacenter, errDc = finder.DatacenterOrDefault(ctx, h.datacenterString); errDc != nil {
-		return fmt.Errorf("Unable to find datacenter %s : %s", h.datacenterString, errDc)
+	if h.datacenter, err = finder.DatacenterOrDefault(ctx, h.datacenterString); err != nil {
+		return fmt.Errorf("Unable to find datacenter %s: %v", h.datacenterString, err)
 	}
 	finder.SetDatacenter(h.datacenter)
 
-	var errN error
-	if h.network, errN = finder.NetworkOrDefault(ctx, h.networkString); errN != nil {
-		return fmt.Errorf("Unable to find network %s : %s", h.networkString, errN)
+	if h.network, err = finder.NetworkOrDefault(ctx, h.networkString); err != nil {
+		return fmt.Errorf("Unable to find network %s: %v", h.networkString, err)
 	}
 
 	go h.main()

--- a/engine/hatchery/vsphere/init.go
+++ b/engine/hatchery/vsphere/init.go
@@ -13,6 +13,11 @@ import (
 
 // InitHatchery create new client for vsphere
 func (h *HatcheryVSphere) InitHatchery(ctx context.Context) error {
+
+	h.user = h.Config.VSphereUser
+	h.password = h.Config.VSpherePassword
+	h.endpoint = h.Config.VSphereEndpoint
+
 	// Connect and login to ESX or vCenter
 	c, errNc := h.newClient(ctx)
 	if errNc != nil {

